bigframes/session/executor.py

```diff
@@ -348,7 +348,10 @@ def export_gcs(
             export_options=dict(export_options),
         )
         job_config = bigquery.QueryJobConfig()
-        bq_io.add_labels(job_config, api_name=f"dataframe-to_{format.lower()}")
+
+        # Note: Ensure no additional labels are added to job_config after this point,
+        # as `add_and_trim_labels` ensures the label count does not exceed 64.
```

Comment:
The repetition of this note at multiple places makes me think - can we not call `start_query_with_client` instead of `bqclient.query` from everywhere? That way any preprocessing (including labels) would be centralized and have only one note lilke this in our code.

Reply:
The logic after bqclient.query in start_query_with_client don't seems to match other functions, so this may not work.

We can make a new function that add labels and execute to replace bqclient.query, what do you think? This is my original thought but I'm not sure if it's necessary to add a new function just for two lines of code.

Reply:
Not a great refactoring but better than leaving repetitive notes IMHO.

Reply:
Or we could make a hook in the `ClientsProvider._create_bigquery_client`, something like:

```python
original_query = bq_client.query

def query (job_config, *args, **kwargs, api_name):
    add_and_trim_labels(job_config, api_name=api_name)
    original_query(job_config, *args, **kwargs)
```

Reply:
I would discourage us from monkey patching like this, though it should work in Python. I think your first instinct of using our central helper function makes the most sense to me.

I don't fully understand what you mean by

> The logic after bqclient.query in start_query_with_client don't seems to match other functions, so this may not work.

We should be able to put `start_query_with_client` in a try/except block. Also, it's probably a bug that we aren't showing that a query job is running here.